llvm/lib/Transforms/Utils/Evaluator.cpp

```diff
@@ -124,6 +125,20 @@ isSimpleEnoughValueToCommit(Constant *C,
   return isSimpleEnoughValueToCommitHelper(C, SimpleConstants, DL);
 }
 
+static std::pair<bool, GlobalValue::LinkageTypes>
+getPruningAndLinkage(const GlobalVariable &GV) {
+  if (const GlobalObject *GO = GV.getBaseObject()) {
+    if (const auto *const T = GO->getMetadata(LLVMContext::MD_repo_ticket)) {
```

Comment:
This is the only auto in this group. Is there a reason?

Reply:
Changed 'auto' to the 'MDNode'.